### PR TITLE
Update hack/build to support mkpr args

### PR DESCRIPTION
**Pull request by smarterclayton**

### CODE REVIEW

File: hack/mkpr [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+repo="${1}"
+ref="${2}"
+
+if [[ -z "${2}" ]]; then
```
Comment (eparis) [inline hack/mkpr]: so you check if there is a `ref`  (which you call a branch) but not if you set a `repo` ?

File: hack/mkpr [hunk 2]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
```
Comment (eparis) [inline hack/mkpr]: shouldn't all scripts be `set -euo pipefail`

File: hack/mkpr (hunk 1 above)
Comment (eparis) [reply hack/mkpr]: why use $2 when you set $ref above?

File: hack/mkpr [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  echo "error: must specify branch" 1>&2
+  exit 1
+fi
+branch_ref="$( git ls-remote https://github.com/${repo}.git refs/heads/${ref} | cut -f 1 -d $'\t' )"
+if [[ -z "${branch_ref}" ]]; then
+  echo "error: no ref found for ${ref}" 1>&2
+  exit 1
+fi
+
+if [[ -n "${3-}" ]]; then
```
Comment (eparis) [inline hack/mkpr]: what is $3?

File: hack/mkpr [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  pr_ref="$( git ls-remote https://github.com/${repo}.git refs/pull/${3}/head | cut -f 1 -d $'\t' )"
+  if [[ -z "${pr_ref}" ]]; then
+    echo "error: no PR ref found for ${3}" 1>&2
+    exit 1
+  fi
+  echo "-base-ref ${ref} -base-sha ${branch_ref} -pull-author smarterclayton -pull-number ${3} -pull-sha ${pr_ref}"
```
Comment (eparis) [inline hack/mkpr]: dude, smarterclayton authors all my PRs.  Now I know why his stats look good and mine like shit.

**Comment by smarterclayton** [comment]
Merged in another PR

**Pull request closed without merging**